File: core/config-manager.lua
-- Bucu Core - Config Manager
-- Handles configuration loading and access

local ConfigManager = {
    _config = {},
    _loaded = false
}

-- Load configuration from file
function ConfigManager:load(configPath)
    configPath = configPath or "config/default-config"
    
    local success, config = pcall(require, configPath)
    
    if not success then
        -- Log error if Logger is available
        if Logger then
            Logger:warn("Failed to load config from: " .. configPath)
            Logger:info("Using empty configuration")
        end
        self._config = {}
        self._loaded = false
        return false
    end
    
    self._config = config
    self._loaded = true
    
    if Logger then
        Logger:info("Configuration loaded from: " .. configPath)
    end
    
    return true
end

-- Get configuration value with dot notation support
-- Example: ConfigManager:get("core.version", "1.0.0")
function ConfigManager:get(key, default)
    if not self._loaded then
        return default
    end
    
    -- Split key by dots
    local keys = {}
    for k in string.gmatch(key, "[^.]+") do
        table.insert(keys, k)
    end
    
    -- Navigate through nested tables
    local value = self._config
    for _, k in ipairs(keys) do
        if type(value) == "table" and value[k] ~= nil then
            value = value[k]
        else
            return default
        end
    end
    
    return value
end

-- Set configuration value (runtime only, not persisted)
function ConfigManager:set(key, value)
    -- Split key by dots
    local keys = {}
    for k in string.gmatch(key, "[^.]+") do
        table.insert(keys, k)
    end
    
    -- Navigate and set value
    local current = self._config
    for i = 1, #keys - 1 do
        local k = keys[i]
        if type(current[k]) ~= "table" then
            current[k] = {}
        end
        current = current[k]
    end
    
    current[keys[#keys]] = value
end

-- Get entire configuration
function ConfigManager:getAll()
    return self._config
end

-- Check if configuration is loaded
function ConfigManager:isLoaded()
    return self._loaded
end

-- Validate configuration structure
function ConfigManager:_validate()
    -- Basic validation
    if type(self._config) ~= "table" then
        if Logger then
            Logger:error("Configuration must be a table")
        end
        return false
    end
    
    -- Validate required sections
    local requiredSections = { "core", "modules", "rateLimit", "cache" }
    for _, section in ipairs(requiredSections) do
        if not self._config[section] then
            if Logger then
                Logger:warn("Missing configuration section: " .. section)
            end
        end
    end
    
    return true
end

return ConfigManager

File: core/config-manager.js
// Bucu Core - Config Manager (JavaScript)
// Handles configuration loading and access

const ConfigManager = {
    _config: {},
    _loaded: false,
    
    // Load configuration from file
    load(configPath = "config/default-config") {
        try {
            // In FiveM, we need to load from Lua side
            // This is a placeholder for JS-side config
            this._config = {};
            this._loaded = false;
            
            if (typeof Logger !== 'undefined') {
                Logger.warn("JS ConfigManager: Config should be loaded from Lua side");
                Logger.info("Using empty configuration");
            }
            
            return false;
        } catch (error) {
            if (typeof Logger !== 'undefined') {
                Logger.warn(`Failed to load config: ${error.message}`);
                Logger.info("Using empty configuration");
            }
            this._config = {};
            this._loaded = false;
            return false;
        }
    },
    
    // Get configuration value with dot notation support
    // Example: ConfigManager.get("core.version", "1.0.0")
    get(key, defaultValue) {
        if (!this._loaded) {
            return defaultValue;
        }
        
        // Split key by dots
        const keys = key.split('.');
        
        // Navigate through nested objects
        let value = this._config;
        for (const k of keys) {
            if (typeof value === 'object' && value !== null && k in value) {
                value = value[k];
            } else {
                return defaultValue;
            }
        }
        
        return value;
    },
    
    // Set configuration value (runtime only, not persisted)
    set(key, value) {
        // Split key by dots
        const keys = key.split('.');
        
        // Navigate and set value
        let current = this._config;
        for (let i = 0; i < keys.length - 1; i++) {
            const k = keys[i];
            if (typeof current[k] !== 'object' || current[k] === null) {
                current[k] = {};
            }
            current = current[k];
        }
        
        current[keys[keys.length - 1]] = value;
    },
    
    // Get entire configuration
    getAll() {
        return this._config;
    },
    
    // Check if configuration is loaded
    isLoaded() {
        return this._loaded;
    },
    
    // Validate configuration structure
    _validate() {
        // Basic validation
        if (typeof this._config !== 'object' || this._config === null) {
            if (typeof Logger !== 'undefined') {
                Logger.error("Configuration must be an object");
            }
            return false;
        }
        
        // Validate required sections
        const requiredSections = ["core", "modules", "rateLimit", "cache"];
        for (const section of requiredSections) {
            if (!(section in this._config)) {
                if (typeof Logger !== 'undefined') {
                    Logger.warn(`Missing configuration section: ${section}`);
                }
            }
        }
        
        return true;
    }
};

// Export for Node.js and FiveM
if (typeof module !== 'undefined' && module.exports) {
    module.exports = ConfigManager;
}

// Make available globally in FiveM
if (typeof global !== 'undefined') {
    global.ConfigManager = ConfigManager;
}
